driver/src/main/java/org/neo4j/driver/internal/NetworkSession.java

```diff
@@ -240,7 +245,7 @@ public void close()
     {
         if ( asyncConnectionFuture != null )
         {
-            return asyncConnectionFuture.thenCombine( new Function<AsyncConnection,InternalFuture<Void>>()
+            return asyncConnectionFuture.thenCompose( new Function<AsyncConnection,InternalFuture<Void>>()
```

Comment:
They can, yeah. I see your point, basically if we close session it will first force-release connection (with RESET message) and then try to close active transaction. Later, however, is not really needed after RESET. I think `else if ( currentAsyncTransactionFuture != null )` branch here is not really required. Will think a bit more about this and probably remove it in a separate PR.